SocialCareCaseViewerApi/V1/Boundary/Requests/PatchWarningNoteRequest.cs

```diff
@@ -16,6 +16,7 @@ public class PatchWarningNoteRequest
         public string ReviewNotes { get; set; }
         public string ManagerName { get; set; }
         public DateTime? DiscussedWithManagerDate { get; set; }
+        public bool DisclosedWithIndividual { get; set; }
```

Comment:
Oooooo, actually just realised does this need to be nullable since you mentioned it's only for adults?

Reply:
I was just about to ask a similar question. Is this adults only? If so, are we being explicit with the validation / errors so that it's required if the resident type is adult, but shouldn't be passed if the resident type is child?

Reply:
Ah, this is true. 👍🏾 
I'll check the frontend request for both. I think the request from a child form would not have a `DisclosedWithIndividual` value since it's not part of the child form. 
And if that is the case, we could update this to allow `DisclosedWithIndividual` to be nullable?

Reply:
@chubberlisk @chrishutchinson I've just checked both requests and realised that we [modify the request](https://github.com/LBHackney-IT/lbh-social-care-frontend/blob/075772eb411318aa816b0704f34293c71a9f6026/pages/people/%5Bid%5D/warning-notes/%5BwarningNoteId%5D/%5B%5B...stepId%5D%5D.tsx#L37) before we send it to the API.

My guess is if `disclosedWithIndividual` was null, the code `disclosedWithIndividual: formData.disclosedWithIndividual === 'Yes',` would return `false`.

What are your thoughts on changing the code in the frontend to check for and allow null, or should this be more granular, and we should check that the person is an adult first and then add the `disclosedWithIndividual` value to the request?

Reply:
My expectation for the input/output would be that the type would be something like this (modelled in typescript just for convenience, but the theory should apply):

```ts
type Data = 
  | {
    residentType: "adult";
    disclosedWithIndividual: boolean;
  }
  | {
    residentType: "child";
    disclosedWithIndividual: never;
  }
```

This is effectively saying:
* If the resident is an adult, the value of `disclosedWithIndividual` **must** be provided, and it **must** be a `boolean`
* If the resident is a child, the value of `disclosedWithIndividual` should **never** be provided

This is more representative of the actual data model.

In the database we may have to store it as a `null` value, but I would expect the API layout should enforce these sort of rules.

Reply:
Thanks @chrishutchinson , 🙌🏾 
I will update the backend (this PR) so that the `disclosedWithIndividual` is nullable and then update the frontend to deal with `disclosedWithIndividual` based on the person.